tests/unit/netascii_test.py

```diff
@@ -0,0 +1,40 @@
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
+
+import io
+import unittest as t
+
+from py3tftp.netascii import Netascii
+
+class NetasciiTest(t.TestCase):
+    octet_to_netascii = {
+        b"\r":     b"\r\x00",
+        b"\n":     b"\r\n",
+        b"te\nst": b"te\r\nst",
+        b"te\rst": b"te\r\x00st",
+        b"\r\r\r": b"\r\x00\r\x00\r\x00",
+        b"\n\n\n": b"\r\n\r\n\r\n",
+    }
+
+    def test_netascii_reader(self):
+        for octet, netascii in self.octet_to_netascii.items():
+            for blksize in [1,2,3,4]:
+                octet_buffer = io.BytesIO(octet)
+                netascii_buffer = b''
```

Comment:
Just FYI for the future - strings, including byte strings, are immutable in Python so every `buffer += b'something'` operation creates a new object, which will incur unnecessary overhead in some situations. Without profiling, I can't say if it applies here, but I'd much prefer to merge this now instead of tweaking it.